src/expr/src/vector_op/trigonometric.rs

```diff
@@ -56,6 +56,81 @@ pub fn atan2_f64(input_x: F64, input_y: F64) -> F64 {
     input_x.0.atan2(input_y.0).into()
 }
 
+// Radians per degree, a.k.a. PI / 180
+static RADIANS_PER_DEGREE: f64 = 0.017_453_292_519_943_295;
+
+// Constants we use to get more accurate results.
+// See PSQL: https://github.com/postgres/postgres/blob/78ec02d612a9b69039ec2610740f738968fe144d/src/backend/utils/adt/float.c#L2024
+static SIN_30: f64 = 0.499_999_999_999_999_94;
+static ONE_MINUS_COS_60: f64 = 0.499_999_999_999_999_9;
+
+// returns the cosine of an angle that lies between 0 and 60 degrees. This will return exactly 1
+// when xi s 0, and exactly 0.5 when x is 60 degrees.
+fn cosd_0_to_60(x: f64) -> f64 {
+    // https://github.com/postgres/postgres/blob/REL_15_2/src/backend/utils/adt/float.c
+    let one_minus_cos_x: f64 = 1.0 - f64::cos(x * RADIANS_PER_DEGREE);
+    1.0 - (one_minus_cos_x / ONE_MINUS_COS_60) / 2.0
+}
+
+// returns the sine of an angle that lies between 0 and 30 degrees. This will return exactly 0 when
+// x is 0, and exactly 0.5 when x is 30 degrees.
+fn sind_0_to_30(x: f64) -> f64 {
+    // https://github.com/postgres/postgres/blob/REL_15_2/src/backend/utils/adt/float.c
+    let sin_x = f64::sin(x * RADIANS_PER_DEGREE);
+    (sin_x / SIN_30) / 2.0
+}
+
+// Returns the sine of an angle in the first quadrant (0 to 90 degrees).
+fn sind_q1(input: f64) -> f64 {
+    // https://github.com/postgres/postgres/blob/REL_15_2/src/backend/utils/adt/float.c
+
+    //  Stitch together the sine and cosine functions for the ranges [0, 30]
+    //  and (30, 90].  These guarantee to return exact answers at their
+    //  endpoints, so the overall result is a continuous monotonic function
+    //  that gives exact results when x = 0, 30 and 90 degrees.
+
+    if input <= 30.0 {
+        sind_0_to_30(input)
+    } else {
+        cosd_0_to_60(90.0 - input)
+    }
+}
+
+#[function("sind(float64) -> float64")]
+pub fn sind_f64(input: F64) -> F64 {
+    // PSQL implementation: https://github.com/postgres/postgres/blob/REL_15_2/src/backend/utils/adt/float.c#L2444
+
+    // Returns NaN if input is NaN or infinite. Different from PSQL implementation.
+    if input.0.is_nan() || input.0.is_infinite() {
+        return f64::NAN.into();
+    }
+
+    let arg1 = input.0 % 360.0;
+    let sign = 1.0;
+
+    let (arg1, sign) = if arg1 < 0.0 {
```

Comment:
`PG`'s implementation uses three independent `if` while `else if` is used here,

is this intentional?

Reply:
Was intentional. I wanted to avoid using `mut` in for `arg1` and `sign`, which is why I use the assignments with the if else combinations.